docker/entrypoint.sh

```diff
@@ -3,17 +3,23 @@
 # Populate the Windows path inside of the wineprefix
 # TODO: This is in here because writing to the registry seems
 # to fail when performed in the Dockerfile itself; investigate
-wine reg ADD 'HKCU\Environment' /v PATH /d 'C:\msvc\bin;C:\msvc\bin\winnt;C:\cmake\bin;C:\windows\system32' /f
+wine reg ADD 'HKCU\Environment' /v PATH /d 'C:\msvc\bin;C:\msvc\bin\winnt;C:\cmake\bin;C:\windows\system32;C:\' /f
 wine reg ADD 'HKCU\Environment' /v INCLUDE /d 'C:\msvc\include;C:\msvc\mfc\include' /f
 wine reg ADD 'HKCU\Environment' /v LIB /d 'C:\msvc\lib;C:\msvc\mfc\lib' /f
 wine reg ADD 'HKCU\Environment' /v TMP /d 'Z:\build' /f
 wine reg ADD 'HKCU\Environment' /v TEMP /d 'Z:\build' /f
 
 # Configure build with CMake
-wine cmake -B build isle -G "NMake Makefiles" $CMAKE_FLAGS
+wine cmake -B build isle -G "Ninja" $CMAKE_FLAGS
```

Comment:
I think the reason we stuck with `NMake Makefiles` was that it has correct dependency tracking.
It did not work well with Ninja.

Reply:
I remember that too, yes. It might not be a concern if the Docker build/script is used only for "one-off" compilation scenarios?

Reply:
Let's assume that

Reply:
The build definitely "works". This `build.sh` does everything isolated from the source repo, throws all intermediate artifacts away, and hence needs to recompile everything every time, yes. But at least a full rebuild is much faster this way. There still seems to be a lot of "critical path moments" in the build, where it still compiles everything one-by-one. That could be improved.

Reply:
I think I know what you mean. The readme says:
> NMake Makefiles is most recommended because it will be immediately compatible with Visual C++ 4.2. For faster builds, you can use Ninja (if you have it installed), however due to limitations in Visual C++ 4.2, you can only build Release builds this way (debug symbols cannot be generated with Ninja).

So if I understand it correctly, Ninja could be used to build all release builds, even incremental ones?

Reply:
incremental builds are the trouble makers when using an older MSVC: files whose includes have changed will not rebuild.

Reply:
Hmm. That's a little weird to me. Ninja is basically just a command runner, so if CMake can generate something that works for NMake, why would that break if all CMake does is (theoretically, I haven't looked at CMake's output myself) generate something equivalent for Ninja?

Reply:
We're talking about 1997-era tooling. Ninja works fine for today's MSVC toolchains.
Until I understand the issue better, I'm assuming there's an external communication channel between nmake and cl.

Reply:
Yeah, I guess it is *that* old. Should we split it into "I just want to build once" and "I want incremental builds" or drop ninja at all? But in that case this PR isn't worth much anymore.